x/epochs/spec/README.md

```diff
@@ -17,63 +17,89 @@ they can easily be signalled upon such events.
 4. **[Keeper](#keeper)**
 5. **[Hooks](#hooks)**
 6. **[Queries](#queries)**
-7. **[Future improvements](#future-improvements)**
+7. **[Downtime Recovery](#downtime-recovery)**
 
 ## Concepts
 
-The purpose of `epochs` module is to provide generalized epoch interface
-to other modules so that they can easily implement epochs without
-keeping own code for epochs.
+The epochs module defines on-chain timers, that execute at fixed time intervals.
+Other SDK modules can then register logic to be executed at the timer ticks.
+We refer to the period in between two timer ticks as an "epoch".
 
-## State
+Every timer has a unique identifier.
+Every epoch will have a start time, and an end time, where `end time = start time + timer interval`.
+On Osmosis mainnet, we only utilize one identifier, with a time interval of `one day`.
+
+The timer will tick at the first block whose blocktime is greater than the timer end time,
+and set the start as the prior timer end time. (Notably, its not set to the block time!)
+This means that if the chain has been down for awhile, you will get one timer tick per block,
+until the timer has caught up.
 
-Epochs module keeps `EpochInfo` objects and modify the information as
-epochs info changes. Epochs are initialized as part of genesis
-initialization, and modified on begin blockers or end blockers.
+## State
 
-### Epoch information type
+Epochs module keeps a single [`EpochInfo`](https://github.com/osmosis-labs/osmosis/blob/b4befe4f3eb97ebb477323234b910c4afafab9b7/proto/osmosis/epochs/genesis.proto#L12) per identifier.
+Its fields are modified at every timer tick. 
+EpochInfos are initialized as part of genesis initialization or upgrade logic,
+and are only modified on begin blockers.
 
 ```protobuf
 message EpochInfo {
-    string identifier = 1;
-    google.protobuf.Timestamp start_time = 2 [
-        (gogoproto.stdtime) = true,
-        (gogoproto.nullable) = false,
-        (gogoproto.moretags) = "yaml:\"start_time\""
-    ];
-    google.protobuf.Duration duration = 3 [
-        (gogoproto.nullable) = false,
-        (gogoproto.stdduration) = true,
-        (gogoproto.jsontag) = "duration,omitempty",
-        (gogoproto.moretags) = "yaml:\"duration\""
-    ];
-    int64 current_epoch = 4;
-    google.protobuf.Timestamp current_epoch_start_time = 5 [
-        (gogoproto.stdtime) = true,
-        (gogoproto.nullable) = false,
-        (gogoproto.moretags) = "yaml:\"current_epoch_start_time\""
-    ];
-    bool epoch_counting_started = 6;
-    reserved 7;
-    int64 current_epoch_start_height = 8;
+  // Identifier is a unique reference to this particular timer.
```

Comment:
I think putting the below would be an approach without too much verbosity. 

```markdown
...
and are only modified on begin blockers.

Each epoch timer has a unique reference identifier. In addition, the individual timers consist of `start_time`, `duration`, `current_epoch`, `current_epoch_start_time`, `current_epoch_start_height`, `epoch_counting_started`, and `reserved`, as seen below:

| Timer name | Description |
| ----------- | ----------- |
| `start_time` | Start time is the time at which the timer first ever ticks. |
| `duration`   | Duration is the time in between epoch ticks. |
| `current_epoch` | How many times has the `start_time` timer 'ticked'. |
| `current_epoch_start_time` | Describes the start time of the current timer interval. | 
| `current_epoch_start_height` | This is the block height at which the current epoch started. (The blockheight at which the timer last ticked) |
| `reserved` | (insert description here, because I don't know what reserved does) |
| `epoch_counting_started` | A boolean that indicates whether the epoch timer has begun yet |
```

The only other thing I would question / recommend is just for `current_epoch_start_time`, maybe an example would help? I found the description of how it worked a bit confusing at the start so please correct my simplified description or even add an extreme-case scenario if that's better.

Reply:
This is a good idea!

I guess something this makes me realize is that it:
* Added a brief list of the fields at the top,
* Cut out the protobuf overhead.

Wdyt about us just deleting all the `gogoproto` protobuf overhead, but keeping the long comments?

Or do you think the succinct descriptions is best for understandability?

(Also ignore the reserved field, thats protobuf specific, about field numbers)

Reply:
From call:
* Lets not duplicate proto here, just link to it
* Lets not care about enumerating all the field names
* Lets only put attention on noting the edge case behavior of relevance